tuta-sdk/rust/src/typed_entity_client.rs

```diff
@@ -0,0 +1,44 @@
+use std::sync::Arc;
+use serde::Deserialize;
+use crate::{ApiCallError, TypeRef};
+use crate::entities::Entity;
+use crate::entity_client::{EntityClient, IdType};
+use crate::instance_mapper::InstanceMapper;
+
+pub struct TypedEntityClient {
+    entity_client: Arc<EntityClient>,
+    instance_mapper: Arc<InstanceMapper>,
+}
+
+impl TypedEntityClient {
+    pub(crate) fn new(
+        entity_client: Arc<EntityClient>,
+        instance_mapper: Arc<InstanceMapper>,
+    ) -> Self {
+        TypedEntityClient {
+            entity_client,
+            instance_mapper,
+        }
+    }
+
+    pub async fn load_typed_element<T: Entity + Deserialize<'static>>(
```

Comment:
I'd keep the name `load` (we should rename the other one too tbh)

Reply:
Fixed.